src/lib.rs

```diff
@@ -42,6 +42,15 @@ pub mod chat_states {
     pub const JOINED: &str = "joined";
 }
 
+#[derive(Debug, Clone)]
+pub struct ChatInput {
```

Comment:
Since `chat` in purple means something specific, how about `MsgInfo` or something alike?

Reply:
done